packages/node/src/utils/cosmos.ts

```diff
@@ -223,18 +228,28 @@ export async function fetchBlocksBatches(
   const blocks = await fetchCosmosBlocksArray(api, blockArray);
   return Promise.all(
     blocks.map(async (blockInfo) => {
-      const txInfos =
+      const decodedTxs =
+        blockInfo.txs.length <= 0
+          ? []
+          : blockInfo.txs.map((tx) => decodeTxRaw(tx));
```

Comment:
If you want to map an empty array there's no need to check if its empty. There's a couple of places in this file it can be applied.

`const decodedTxs = blockInfo.txs.map((tx) => decodeTxRaw(tx));`